examples/hitl/rearrange_v2/object_state_manipulator.py

```diff
@@ -0,0 +1,172 @@
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, Tuple, cast
+
+from habitat_llm.sims.collaboration_sim import CollaborationSim
+from habitat_llm.tools.motor_skills.object_states.oracle_clean_skills import (
+    OracleCleanInPlaceSkill,
+)
+from habitat_llm.tools.motor_skills.object_states.oracle_fill_skills import (
+    OracleFillInPlaceSkill,
+)
+from habitat_llm.tools.motor_skills.object_states.oracle_object_state_skill import (
+    OracleObjectStateInPlaceSkill,
+)
+from habitat_llm.tools.motor_skills.object_states.oracle_pour_skills import (
+    OraclePourInPlaceSkill,
+)
+from habitat_llm.tools.motor_skills.object_states.oracle_power_skills import (
+    OraclePowerOffInPlaceSkill,
+    OraclePowerOnInPlaceSkill,
+)
+from world import World
+
+from habitat.sims.habitat_simulator import sim_utilities
+from habitat.sims.habitat_simulator.object_state_machine import (
+    ObjectStateSpec,
+    set_state_of_obj,
+)
+
+BooleanActionMap = Dict[bool, OracleObjectStateInPlaceSkill]
+
+
+@dataclass
+class BooleanAction:
+    state_spec: ObjectStateSpec
+    current_value: bool
+    target_value: bool
+    available: bool
+    enabled: bool
+    error: Optional[str]
+
+
+class ObjectStateManipulator:
+    """
+    Helper class for reading and writing object states for a given world, on behalf of a given agent.
+    Requires the 'habitat_llm' external module.
+    """
+
+    def __init__(
+        self,
+        sim: CollaborationSim,
+        agent_index: int,
+        world: World,
+        maximum_distance: float,
+    ):
+        self._sim = sim
+        self._agent_index = agent_index
+        self._maximum_distance = maximum_distance
+        self._world = world
+
+        self._boolean_state_action_map: Dict[str, BooleanActionMap] = {
+            "is_clean": {True: OracleCleanInPlaceSkill},
+            "is_filled": {
+                True: OracleFillInPlaceSkill,
+                False: OraclePourInPlaceSkill,
```

Comment:
Add a quick comment about the relationship b/w skill and state. I think you mean `Skill1/Skill2` will cause this state to switch to `True/False`